src/cmd/describegpt.rs

```diff
@@ -167,10 +271,19 @@ fn get_completion(api_key: &str, messages: &serde_json::Value, args: &Args) -> C
         return fail!("Error: Invalid model.");
     }
 
+    // Get max_tokens from prompt file if --prompt-file is used
+    let max_tokens = match args.flag_prompt_file.clone() {
+        Some(prompt_file) => {
+            let prompt_file = get_prompt_file(args)?;
+            prompt_file.tokens
+        }
+        None => args.flag_max_tokens,
+    };
```

Comment:
@jqnatividad Since `--max-tokens` has a default of 50, how can we check if the option itself is being used?

I'm trying to make it so that if `--max-tokens` is explicitly set then it overrides the prompt file `tokens` value like you mentioned [in the issue](https://github.com/jqnatividad/qsv/issues/1085#:~:text=the%20tokens%20attribute%20will%20be%20used%20unless%20%2D%2Dmax%2Dtokens%20option%20is%20set%20explicitly%20when%20invoking%20describegpt.).

Reply:
@rzmk , it's a bit hacky, but this is what I'd do:

```rust
pub fn run(argv: &[&str]) -> CliResult<()> {
  let args: Args = util::get_args(USAGE, argv)?;
  let argv_defaults = &["describegpt"];
  let args_defaults: Args = util::get_args(USAGE, argv_defaults)?;
```
so `args_defaults.flag_max_tokens` will contain the default value specified in the USAGE text, which you can use later in the code to see if it was changed.  This approach gives us the flexibility to insulate ourselves from future changes to the USAGE text as defaults, additional args/options are added.

Reply:
@jqnatividad Getting an error this way:

```console
$ ./target/debug/qsv describegpt NYC_311_SR_2010-2020-sample-1M.csv --tags --max-tokens 100
Invalid arguments.

Usage:
    qsv describegpt [options] [<input>]
    qsv describegpt --help
```

Also another two ways that may work are:

- Looping through `argv` for `--max-tokens` and getting the subsequent value.
- Making `--max-tokens` optional so we can use `is_some()`.

Both methods don't make `args_default` though which could be useful in the future.

Reply:
My mistake @rzmk , the first element of argv is the name of the binary:

```rust
pub fn run(argv: &[&str]) -> CliResult<()> {
  let args: Args = util::get_args(USAGE, argv)?;
  // simulate invoking describegpt with just a stdin input, and no options set, so we get the defaults
  let argv_defaults = &[argv[0], "describegpt", "-"];
  let args_defaults: Args = util::get_args(USAGE, argv_defaults)?;
```

Reply:
Still getting the same error @jqnatividad.

Reply:
Hhmmm....

It's working for me:
```
$ cargo t describegpt -F lite

...

$ target/debug/qsvlite describegpt C:\Users\joeln\Downloads\NYC_311_SR_2010-2020-sample-1M.csv --tags --max-tokens 100
Generating stats from \\?\C:\Users\joeln\Downloads\NYC_311_SR_2010-2020-sample-1M.csv using qsv stats --everything...
Generating frequency from \\?\C:\Users\joeln\Downloads\NYC_311_SR_2010-2020-sample-1M.csv using qsv frequency...
Interacting with OpenAI API...

Generating tags from OpenAI API...
Received tags completion.
uniquekey, createddate, closeddate, agency, agencyname, complainttype, descriptor, locationtype, incidentzip, incidentaddress, streetname, crossstreet1, crossstreet2, intersectionstreet1, intersectionstreet2, addressType, city, landmark, facilitytype, status, duedate, resolutiondescription, resolutionactionupdateddate, communityboard, bbl, borough, xcoordinatestateplane, ycoordinatestateplane, opendatachanneltype, parkfacilityname, park
```

Reply:
Somehow it works now?! 🎉

Reply:
@jqnatividad What if a user specifies `--max-tokens 50` then how can I differentiate between default? This is mainly what I'm considering because a user may have `7000` in their prompt file for `tokens` but then also set `--max-tokens 50` (i.e. for testing the command). How can I prioritize `--max-tokens` with `args.flag_max_tokens` over the prompt file value if `args.flag_max_tokens == args.flag_max_tokens`?